packages/compass/src/setup-hadron-distribution.ts

```diff
@@ -32,6 +32,26 @@ if (
   // type `browser` indicates that we are in the main electron process
   process.type === 'browser'
 ) {
+  if (process.platform === 'linux') {
+    // For Linux users with drivers that are avoided by Chromium we disable the
+    // GPU check to attempt to bypass the disabled WebGL settings.
+    app.commandLine.appendSwitch('ignore-gpu-blacklist', 'true');
+
+    /**
+     * Default password store detection in Chromium relies on pre-defined set of
+     * values for XDG_CURRENT_DESKTOP env var. Even if the store is installed
+     * correctly, Chromium might fail to detect that. To try to work around that
+     * we explicitly set password-store to gnome-libsecret for any linux
+     * platform (we list gnome-keyring as a dependency for all compass
+     * installers on linux)
+     *
+     * @see {@link https://github.com/microsoft/vscode/issues/185212#issuecomment-1593271415}
+     */
+    if (app.commandLine.hasSwitch('password-store') === false) {
+      app.commandLine.appendSwitch('password-store', 'gnome-libsecret');
```

Comment:
Are there any other factors we should be taking into account when setting this defailt, e.g. specific values of $XDG_CURRENT_DESKTOP or whether the `gnome-keyring` binary is in PATH/whether a `libsecret.so` is available on the current system?

Reply:
That's a good point! I was thinking about checking that XDG_ var is not in the "supported" list, but it seemed like a mess to maintain. I like the idea of checking for libsecret!

One thing I want to do first is to check what exactly happens when you set this in the environment where it's not supported. If it's not different from not setting it at all I think, just from the perspective of having less logic to maintain, I would prefer not to do any extra checks, but please tell me if this doesn't sound right to you 🙂